bundles/org.openhab.binding.qolsysiq/src/main/java/org/openhab/binding/qolsysiq/internal/handler/QolsysIQZoneHandler.java

```diff
@@ -94,17 +94,38 @@ protected void updateZone(Zone zone) {
     }
 
     protected void zoneActiveEvent(ZoneActiveEvent event) {
-        if (event.zone.zoneId == zoneId()) {
+        if (event.zone.zoneId == getZoneId()) {
             updateZoneStatus(event.zone.status);
         }
     }
 
     protected void zoneUpdateEvent(ZoneUpdateEvent event) {
-        if (event.zone.zoneId == zoneId()) {
+        if (event.zone.zoneId == getZoneId()) {
             updateZone(event.zone);
         }
     }
 
+    private void initializeZone() {
+        Bridge bridge = getBridge();
+        if (bridge != null) {
+            BridgeHandler handler = bridge.getHandler();
+            if (handler instanceof QolsysIQPartitionHandler) {
```

Comment:
There is still a case not covered (even if it should  not happen). You can cover it using this code:
```
Bridge bridge = getBridge();
BridgeHandler handler = bridge == null ? null : bridge.getHandler();
if (bridge != null && handler instanceof QolsysIQPartitionHandler) {
    ...
} else {
    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_UNINITIALIZED);
}
```